CameraOverlay/overlay_all_stats.py

```diff
@@ -0,0 +1,160 @@
+import os
+import time
+from overlay import Overlay
+from PIL import Image, ImageDraw, ImageFont
+import topics
+
+
+class OverlayAllStats(Overlay):
+
+	def __init__(self):
+		super(OverlayAllStats, self).__init__()
+
+		self.prev_time = 0
+
+		self.font_path = os.path.join(os.path.dirname(__file__), 'FreeSans.ttf')
+		self.bottom_text_height = 70
+		self.bottom_text_font = ImageFont.truetype(self.font_path, self.bottom_text_height)
+		self.top_text_height = 45
+		self.top_text_font = ImageFont.truetype(self.font_path, self.top_text_height)
+		self.top_box_height = 80
+		self.top_box_width = self.width
+
+	def on_connect(self, client, userdata, flags, rc):
+		print("Connected with rc: " + str(rc))
+		# Subscribed topics
+		client.subscribe(topics.DAS.start)
+		client.subscribe(topics.DAS.data)
+		client.subscribe(topics.DAS.stop)
+		client.subscribe(topics.PowerModel.recommended_sp)
+		client.subscribe(topics.PowerModel.predicted_max_speed)
+		client.subscribe(topics.PowerModel.plan_name)
+
+		# Add static text
+		img = Image.new('RGBA', (self.width, self.height))
+		draw = ImageDraw.Draw(img)
+		draw.rectangle(((0, 0), (self.top_box_width, self.top_box_height)), fill='black')
+		draw.text((0, (self.top_box_height - self.top_text_height) / 2 + 8), 'T: ', font=self.top_text_font,
+		          fill='white')
+		draw.text((256, (self.top_box_height - self.top_text_height) / 2 + 8), 'ZDL: ', font=self.top_text_font,
```

Comment:
Might be worth having  `(self.top_box_height - self.top_text_height) / 2 + 8` as a private class variable?

Reply:
yes, agreed. there are just so many things that can be optimised. but i simply wanted it to work with the refactored code. maybe optimizations can be done in the next iteration